cache-config/tm-health-client/config/config.go

```diff
@@ -0,0 +1,256 @@
+			if strings.HasPrefix(v, "TO_") {
+				sf := strings.Split(v, "=")
+				if len(sf) == 2 {
+					if sf[0] == "TO_URL" {
+						// parse the url after trimming off any surrounding double quotes
+						cfg.TOUrl = strings.Trim(sf[1], "\"")
+						to_url_found = true
+					}
+					if sf[0] == "TO_USER" {
+						// set the TOUser after trimming off any surrounding quotes.
+						cfg.TOUser = strings.Trim(sf[1], "\"")
+						to_user_found = true
+					}
+					// set the TOPass after trimming off any surrounding quotes.
+					if sf[0] == "TO_PASS" {
+						cfg.TOPass = strings.Trim(sf[1], "\"")
+						to_pass_found = true
+					}
+				}
+			}
+		}
+	}
+	if !to_url_found && !to_user_found && !to_pass_found {
+		return errors.New("failed to retrieve one or more TrafficOps credentails")
+	}
+
+	return nil
+}
+
+func GetConfig() (Cfg, error, bool) {
+	var err error
+	var configFile string
+	var logLocationErr = log.LogLocationStderr
+	var logLocationDebug = log.LogLocationNull
+	var logLocationInfo = log.LogLocationNull
+	var logLocationWarn = log.LogLocationNull
+
+	configFilePtr := getopt.StringLong("config-file", 'f', DefaultConfigFile, "full path to the json config file")
+	logdirPtr := getopt.StringLong("logging-dir", 'l', DefaultLogDirectory, "directory location for log files")
+	helpPtr := getopt.BoolLong("help", 'h', "Print usage information and exit")
+	verbosePtr := getopt.CounterLong("verbose", 'v', `Log verbosity. Logging is output to stderr. By default, errors are logged. To log warnings, pass '-v'. To log info, pass '-vv', debug pass '-vvv'`)
+
+	getopt.Parse()
+
+	if configFilePtr != nil {
+		configFile = *configFilePtr
+	} else {
+		configFile = DefaultConfigFile
+	}
+
+	var logfile string
+
+	logfile = *logdirPtr + "/" + DefaultLogFile
+
+	logLocationErr = logfile
+
+	if *verbosePtr == 1 {
+		logLocationWarn = logfile
+	} else if *verbosePtr == 2 {
+		logLocationInfo = logfile
+		logLocationWarn = logfile
+	} else if *verbosePtr == 3 {
+		logLocationInfo = logfile
+		logLocationWarn = logfile
+		logLocationDebug = logfile
+	}
+
+	if help := *helpPtr; help == true {
+		Usage()
+		return Cfg{}, nil, true
+	}
+
+	lcfg := LogCfg{
+		LogLocationDebug: logLocationDebug,
+		LogLocationErr:   logLocationErr,
+		LogLocationInfo:  logLocationInfo,
+		LogLocationWarn:  logLocationWarn,
+	}
+
+	if err := log.InitCfg(&lcfg); err != nil {
+		return Cfg{}, errors.New("Initializing loggers: " + err.Error() + "\n"), false
+	}
+
+	cfg := Cfg{
+		TrafficMonitors: make(map[string]bool, 0),
+	}
+
+	if err = LoadConfig(&cfg, configFile); err != nil {
+		return Cfg{}, errors.New(err.Error() + "\n"), false
+	}
+
+	if err = readCredentials(&cfg); err != nil {
+		return cfg, err, false
+	}
+
+	err = GetTrafficMonitorsStatus(&cfg)
+	if err != nil {
+		return cfg, err, false
+	}
+
+	return cfg, nil, false
+}
+
+func GetTrafficMonitorsStatus(cfg *Cfg) error {
+	u, err := url.Parse(cfg.TOUrl + "/api/3.0/servers?type=RASCAL&status=ONLINE")
```

Comment:
Am I misreading this, or is the path here never used? I was worried about hard-coding the `3.0`, but it looks like it isn't used, it's just to create a valid URL to parse?

It looks like `url.Parse("?type=RASCAL&status=ONLINE")` will work: https://play.golang.org/p/gxKCH7wCZZ8

Or you could also:
```
qry := url.Values{}
qry.Add("type", "RASCAL")
qry.Add("status", "ONLINE")
```

Reply:
Fixed, yeah I had just pasted that path in.